common/utils.js

```diff
@@ -1173,6 +1228,15 @@
         }
     }])
 
+    .factory("ConfigUtils", [function() {
+        function getConfigJSON() {
```

Comment:
I thought this function would take care of checking whether chaiseConfig is defined or not. It's easier to have that in here, instead of always returning the default here and then delegating the task of checking the existence of chaiseConfig to the caller.

Also I thought we want one singular place to attach the default variables for each individual attributes. Assume `chaiseConfig` is defined but `chaiseConfig.attribute` is not. Then this function should actually attach the default value of `attribute` to the `chaiseConfig`(Although this is only valid for attributes that their default value is a truthy value like a non-empty string, `true`, or a number). 

If you do these then we don't need to check for existence of chaiseConfig in different places.

Reply:
Moved the check for 'chaiseConfig != "undefined"' to the function getConfigJSON(), and removed the redundant checks from all files.
A few places require the check (e.g. Cases where we do not have a default value) so I have kept them the same.